subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/BuildCacheStepTest.groovy

```diff
@@ -91,6 +91,9 @@ class BuildCacheStepTest extends StepSpec<IncrementalChangesContext> implements
     }
 
     def "executes work and stores in cache on cache miss"() {
```

Comment:
:x: We need a test here for the new functionality.

Reply:
Added in 21669018f8237d169fbda2342cbe0201152dc68e.